llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -6442,27 +6442,27 @@ unsigned CombinerHelper::getFPMinMaxOpcForSelect(
   case CmpInst::FCMP_OGT:
   case CmpInst::FCMP_OGE:
     if (VsNaNRetVal == SelectPatternNaNBehaviour::RETURNS_OTHER)
-      return TargetOpcode::G_FMAXNUM;
+      return TargetOpcode::G_FMAXNUM_IEEE;
```

Comment:
These are not freely interchangeable opcodes. SelectPatternNaNBehaviour does not cover the snan behavior

Reply:
I checked how SDISel works on this, it seems the difference is that in GISel we are being more conservative because of -0 vs 0. 

Now that we are fixing G_FMAXNUM/G_FMINNUM to be similar to IEEE variants, I guess we can just relax the restriction on -0.

Reply:
I would hold off on this whole effort until #112852 is in

Reply:
Or more specifically, the follow up work to adjust the uses